src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -300,33 +305,87 @@ private static void SetProviderConfiguration(string strConfig, EventPipeConfigur
                     {
-                        level = 0;
+                        case 0: // provider
+                            providerName = components[i];
+                            break;
+                        case 1: // keywords
+                            // We use a try/catch block here because ulong.TryParse won't accept 0x at the beginning
+                            // of a hex string.  Thus, we either need to conditionally strip it or handle the exception.
+                            // Given that this is not a perf-critical path, catching the exception is the simpler code.
+                            try
+                            {
+                                keywords = Convert.ToUInt64(components[i], 16);
+                            }
+                            catch { }
+                            break;
+                        case 2: // level
+                            uint.TryParse(components[i], out level);
+                            break;
+                        case 3: // parameter
+                            var dict = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+                            string[] parameters = components[i].Split(
+                                FilterDataDelimiter,
+                                StringSplitOptions.RemoveEmptyEntries);
+
+                            foreach (string parameter in parameters)
+                            {
+                                string[] pair = parameter.Split(
+                                    '=',
+                                    StringSplitOptions.RemoveEmptyEntries);
+                                if (pair.Length == 2)
+                                {
+                                    string key = pair[0].Trim();
+                                    string value = pair[1].Trim();
+
+                                    // Sanitize user input.
+                                    if (!string.IsNullOrWhiteSpace(key) && !string.IsNullOrWhiteSpace(value))
+                                        dict[key] = value;
+                                }
+                            }
+
+                            // TODO: No Linq
+                            // filterData = String.Join(
+                            //     FilterDataDelimiter,
+                            //     dict.Select(pair => $"{pair.Key}={pair.Value}"));
+                            StringBuilder sb = new StringBuilder();
+                            foreach (var pair in dict)
+                                sb.Append($"{pair.Key}={pair.Value};");
+                            if (sb.Length > 0)
+                                sb.Length--;
+                            filterData = sb.ToString();
+
+                            // Size of filterData cannot exceed 1024 bytes.
+                            if (filterData.Length * sizeof(char) > 1024)
+                                filterData = null; // TODO: Ignore?
```

Comment:
I don't think we actually have a 1024 limit in eventPipe (ETW does, but we don't need to have it).  I would recomment simply removing it.    In any case, we should not check for it here.